shell/server/drivers/mail.js

```diff
@@ -116,7 +119,13 @@ Meteor.startup(function () {
           // there will be an nginx frontend verifying hostnames anyway. Grain public IDs are
           // globally unique anyway, so an e-mail meant for another server presumably won't match
           // any ID at this one anyway.
-          return deliverTo.slice(0, deliverTo.indexOf("@"));
+          const match = PUBLIC_ID_REGEX.exec(deliverTo);
+          if (match) {
+            return match[1];
+          }
+          else {
+            throw new Error("Invalid recipient e-mail address: " + deliverTo);
```

Comment:
We could also just return `null` which would then more or less ignore this recipient and log an error.